specification/cognitiveservices/data-plane/AzureOpenAI/authoring/stable/2022-12-01/azureopenai.json

```diff
@@ -0,0 +1,1965 @@
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ModelList"
+            }
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Getting all models owned by account.": {
+            "$ref": "./examples/get_models.json"
+          }
+        }
+      }
+    },
+    "/models/{model_id}": {
+      "get": {
+        "tags": [
+          "Models:"
+        ],
+        "summary": "Gets details for the model specified by the given modelId.",
+        "operationId": "Models_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "model_id",
+            "description": "The identifier of the model.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionQueryParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Model"
+            }
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Getting a base model.": {
+            "$ref": "./examples/get_model_base.json"
+          },
+          "Getting a fine-tuned model.": {
+            "$ref": "./examples/get_model_finetune.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Capabilities": {
+      "title": "Capabilities",
+      "description": "The capabilities of a base or fine tune model.",
+      "type": "object",
+      "properties": {
+        "fine_tune": {
+          "description": "A value indicating whether a model can be used for fine tuning.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "inference": {
+          "description": "A value indicating whether a model can be deployed.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "completion": {
+          "description": "A value indicating whether a model supports completion.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "embeddings": {
+          "description": "A value indicating whether a model supports embeddings.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "scale_types": {
+          "description": "The supported scale types for deployments of this model.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ScaleType"
+          },
+          "readOnly": true
+        }
+      }
```

Comment:
It looks like this schema is only used for output -- never sent in a request body -- so `readOnly: true` has no purpose on these properties and is best removed. I believe there are other output-only schemas that likewise should have no `readOnly: true` properties.

Reply:
removed required for all response only schemas and added required in response only data when the property is always written by the service